#! /bin/sh -e

# DP: capi20/capidyn.c: Search for libcapi20.so.2

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- capi20/capidyn.c~	2004-01-16 16:27:11.000000000 +0100
+++ capi20/capidyn.c	2004-09-10 16:22:10.000000000 +0200
@@ -26,7 +26,7 @@
 #include <dlfcn.h>
 
 
-#define LIBCAPI	"libcapi20.so"
+#define LIBCAPI	"libcapi20.so.3"
 static int loadlib(void);
 
 /* ---------------------------------------------------------------------- */
